### PR TITLE
Multi-hop payments

### PR DESCRIPTION
**Pull request by metalicjames**
This PR adds multi-hop payment support to Lit. It is possible to route from any origin coin type to a different coin type, thus enabling cross-chain swaps. Routing fees can be implemented by setting an exchange rate between the same coin type other than 1.

Adds the `paymultihop` command to `lit-af` e.g: 
`paymultihop ln1up6k8vxt7vjqcnyzq08e5w6gv8ckr466fcswl3 28 1 250000` 
to pay `ln1up6k8vxt7vjqcnyzq08e5w6gv8ckr466fcswl3` 250000 satoshis of coin type 1 to be delivered as coin type 28.

At present preimages are generated randomly.

### CODE REVIEW

**Comment by treyzania** [comment]
>At present preimages are generated randomly and are not saved between restarts. That should probably be rectified before this PR is merged.

As in they're not serialized to disk?

**Comment by metalicjames** [comment]
That is correct.

File: qln/break.go [hunk 1]
```diff
@@ -40,6 +40,21 @@ func (nd *LitNode) BreakChannel(q *Qchan) error {
...
 	// set delta to 0... needed for break
 	q.State.Delta = 0
+
+	for i, h := range q.State.HTLCs {
+		if !h.Cleared && h.Clearing {
+			q.State.HTLCs[i].Clearing = false
+		}
+	}
+
+	if q.State.InProgHTLC != nil {
```
Comment (delbonis) [inline qln/break.go]: You don't need to do a branch here, if it's already nil then you won't do anything.

**Comment by delbonis** [comment]
@metalicjames There's a few merge conflicts here, can you take a look at them?

**Comment by gertjaap** [comment]
Implements #242 #243 #244

**Pull request merged**